pkg/machine/qemu/machine.go

```diff
@@ -517,14 +514,25 @@ func (v *MachineVM) Start(name string, _ machine.StartOptions) error {
 	if err != nil {
 		return err
 	}
-
 	fd, err := qemuSocketConn.(*net.UnixConn).File()
 	if err != nil {
 		return err
 	}
 
+	dnr, err := os.OpenFile("/dev/null", os.O_RDONLY, 0755)
+	if err != nil {
+		return err
+	}
+	dnw, err := os.OpenFile("/dev/null", os.O_WRONLY, 0755)
+	if err != nil {
+		return err
+	}
+
+	defer dnr.Close()
```

Comment:
the defer statement should be after the file is opened after line 525. If the second open file fails it will leak this fd otherwise.
same for the qemuSocketConn fd.